### ISSUE TITLE
Chrome Extentions AdBlock & Adblock PLUS blocking BuddyPress Followers Button

### ISSUE THREAD

**Issue by shay1383** [r-a-y/buddypress-followers#92]
Hi
First, your plugin is awesome.
The Problem: I have noticed that when AdBlock & Adblock PLUS are activated on my site the follow buttons get disappear.
And when I deactivate the extentions the follow button show up again.
This problem is not matter of theme bug, because I have checked it on my premium theme, but also on WP default themes, as Twenty Seventeen
Please help me and fix it
Regards,
Shay.

**Comment by renatonascalves** [comment]
I can't confirm that but it might be possible because of the usage of the word `follow`. Something to investigate.

**Comment by shay1383** [comment]
I Agree

**Comment by r-a-y** [comment]
Yes, it is due to the usage of the CSS class `follow` and an adblocking filter list hiding elements matching this class.

Unfortunately, this is a user decision by enabling probably a Social Media adblocking filter list that hides any element matching the CSS class, `follow`.  Try looking at what filter lists you have enabled in Adblock Plus or uBlock Origin and narrow down which one is blocking it.  For this reason, I don't think this is necessarily a problem with BP Follow per se.

If I changed this CSS class, other theme developers that might have styled this button will no longer work.

**Comment by shay1383** [comment]
Do you suggest contactnig AdBlock developers? If so, what should I tell them?

**Comment by renatonascalves** [comment]
I'd recommend adding a filter to whitelist it. If that's possible. If it's not, contact the dev and link to this issue.

**Comment by mrjarbenne** [comment]
Looks like you can add your own filters. https://adblockplus.org/forum/viewtopic.php?f=2&t=42013

I would look at the Exception Rules: Example 1 section for details

https://adblockplus.org/en/filter-cheatsheet

**Comment by r-a-y** [comment]
Well, I've narrowed it down to Fanboy's Annoyance List.

It targets the `follow-button` class, which BP Follow uses.  If the Fanboy Annoyance filter list meant to target another, specfic button from Twitter, then a report should be made here:

```
! Please report any unblocked content or problems by email or in our forums
! Email: easylist.subscription@gmail.com
! Homepage: https://easylist.to/
! Forums: https://forums.lanik.us/
```

**Comment by shay1383** [comment]
Does anyone has a solution to how I can disable the follow removal to all my users?
Maybe for exmaple rename follow-button

**Comment by shay1383** [comment]
Hi @r-a-y 
Can you please guide me on how to change 'follow' class in your plugin?
Thanks,
Shay

**Comment by rekmla** [comment]
Hello, we are running into this issue with Adblock Plus. Any plan to address this in the plugin?

**Comment by rekmla** [comment]
@ray, I do like the idea of prefixing the class, but you make good points about this proposed change.

I don't want to cause more problems than are being solved.

**Comment by renatonascalves** [comment]
@r-a-y I'd recommend not changing this. You will create possible problems because of outside tool. The problem is not the plugin.
